src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSClient.java

```diff
@@ -50,29 +48,96 @@
+     *
+     * @param tspDomain         Domain where the Tenant Security Proxy is running.
+     * @param apiKey            Key to use for requests to the Tenant Security Proxy.
+     * @param requestThreadSize Number of threads to use for fixed-size web request thread pool
+     * @param aesThreadSize     Number of threads to use for fixed-size AES operations threadpool
+     * @throws Exception If the provided domain is invalid.
+     */
+    public TenantSecurityKMSClient(String tspDomain, String apiKey, int requestThreadSize,
+            int aesThreadSize) throws Exception {
+        this(tspDomain, apiKey, requestThreadSize, aesThreadSize,
+                SecureRandom.getInstance("NativePRNGNonBlocking"));
+    }
+
+    /**
+     * Constructor for TenantSecurityKMSClient class that allows call to provide web request and AES
+     * operation thread pool sizes. Uses the SecureRandom NativePRNGNonBlocking instance for random
+     * number generation.
+     *
+     * @param tspDomain         Domain where the Tenant Security Proxy is running.
+     * @param apiKey            Key to use for requests to the Tenant Security Proxy.
+     * @param requestThreadSize Number of threads to use for fixed-size web request thread pool
+     * @param aesThreadSize     Number of threads to use for fixed-size AES operations threadpool
+     * @param timeout           Request read and connect timeout in ms. Applies to TSP KMS calls as
+     *                          well.
+     * 
+     * @throws Exception If the provided domain is invalid.
+     */
+    public TenantSecurityKMSClient(String tspDomain, String apiKey, int requestThreadSize,
+            int aesThreadSize, int timeout) throws Exception {
+        this(tspDomain, apiKey, requestThreadSize, aesThreadSize,
+                SecureRandom.getInstance("NativePRNGNonBlocking"), timeout);
+    }
+
+    /**
+     * Constructor for TenantSecurityKMSClient class that allows for modifying the random number
+     * generator used for encryption. Sets a default connect and read timeout of 20s.
+     *
+     * @param tspDomain         Domain where the Tenant Security Proxy is running.
+     * @param apiKey            Key to use for requests to the Tenant Security Proxy.
+     * @param requestThreadSize Number of threads to use for fixed-size web request thread pool
+     * @param aesThreadSize     Number of threads to use for fixed-size AES operations threadpool
+     * @param randomGen         Instance of SecureRandom to use for PRNG when performing encryption
+     *                          operations.
+     * @throws Exception If the provided domain is invalid or the provided SecureRandom instance is
+     *                   not set.
+     */
+    public TenantSecurityKMSClient(String tspDomain, String apiKey, int requestThreadSize,
+            int aesThreadSize, SecureRandom randomGen) throws Exception {
+        this(tspDomain, apiKey, requestThreadSize, aesThreadSize, randomGen, 20000);
```

Comment:
20 seconds is the default timeout on the Google client internally, right?

Reply:
Yes, as far as I could tell.